charts/external-dns/templates/clusterrole.yaml

```diff
@@ -1,12 +1,12 @@
 {{- if .Values.rbac.create -}}
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
+kind: {{ .Values.namespaced | ternary "Role" "ClusterRole" }}
```

Comment:
Is this correct? Or should it be a `ClusterRole` but only a `RoleBinding`?

Reply:
Yes. 

1. I do not have cluster-wide permissions, I'm not able to setup cluster roles.
2. In case, i have install external-dns multiple times on different namespaces, there would be a conflict between multiple installation, since cluster-role isn't namespaced scope. 
3. If the helm chart offers a namespaced option, I would assume no cluster wide objects will be installed.

Reply:
That makes sense but in which case we need the namespace field set explicitly (and for the binding.

Reply:
I though the same. But there is no namespace field. (Don't get confused by with with subjects fields).

RoleBindings are already namespaced and can not refer a Role from a different namespace. RoleBindings always expects Roles in the same namespaces, except roleRef.kind is a ClusterRole.

Reply:
Thanks for the clarification.